### ISSUE TITLE
Game crashes when Warlock chosen is killed on retaliation mission

### ISSUE THREAD

**Issue by kaphula** [long-war-2/lwotc#1624]
Game crashes on ironman when chosen is killed. Only LWOTC2 mod from Steam and its dependencies installed. OS: Linux.

On the save file you should be able to double barrel dead the chosen on the same turn: 

[save_IRONMAN- Campaign 546.zip](https://github.com/long-war-2/lwotc/files/13325062/save_IRONMAN-.Campaign.546.zip)

**Comment by Tvrdoglavi** [comment]
I have the same issue on first time killing the chosen in the game. Form it was one of those "Unhindered Operations" missions.

**Comment by nusendra** [comment]
I have alot of bad experiences on playing ironman. Im pretty sure somehwre said that they don't recommend ironman on LWOTC

**Comment by richard-melvin** [comment]
on windows and latest beta, the provided save doesn't crash, either when the warlock dies, or the mission completes. nothing really unusual in the log. One soldier is missing a headshot (placeholder img instead).

**Comment by Tvrdoglavi** [comment]
The timing of the crash is interesting. I assume that the game calculates the result of an attack before animating it. The game crashes the moment I click on the attack that would kill the chosen. It is the same with both of the runs where I have this crash. It makes it a game breaking bug since it is not possible to avoid the chosen.
I'm attaching my two saves. I hope that helps. 
[Save Files.zip](https://github.com/long-war-2/lwotc/files/13491687/Save.Files.zip)

**Comment by Tvrdoglavi** [comment]
Yes, it's there is something about the large genelab map that crashes the game on Linux. That was a relatively easy fix to remove the map from the game. Unfortunately, removing the chosen is not really an option here. I've enjoyed the mod quite a bit until the point where it crashes. I hope to be able to play it at some point.

**Comment by Tedster59** [comment]
do you have the log files from these crashes too?

**Comment by kaphula** [comment]
Could you instruct me where they are located, if you know?

**Comment by Tvrdoglavi** [comment]
My understanding is that .dmp is the only crash file created on Linux. No crash logs are generated.

**Comment by mdbowler** [comment]
I have exactly the same issue with the first time killing a chosen on a retaliation mission. As soon as it dies, either by being shot or burning, the game crashes.

Platform: Linux, not Ironman

Dump and save attached.

Save: [lwotc_retaliation_crash.zip](https://github.com/long-war-2/lwotc/files/13805134/lwotc_retaliation_crash.zip)

Crash dump: 
[lwotc_retaliation_dump.zip](https://github.com/long-war-2/lwotc/files/13805136/lwotc_retaliation_dump.zip)

**Comment by Iridar** [comment]
We can't actually analyze crash dumps, only the game's Launch.log.

**Comment by Foldi-One** [comment]
I've had this with killing a different Chosen, crash with SIGSEV(11) and a complaint about /lib/private_symbol_hack.so on Linux. I can confirm having experienced this bug myself now that the same save works on windows and will pass the killing of the chosen. (Which isn't a shock given the error, but nice to prove).

**Comment by Lightsockie** [comment]
Just had this happen to me killing the Assassin on a "retaliation" mission. I'm like fairly certain I've killed plenty of chosen on my last lwotc run (circa start of 2023). Could be a regression, maybe? 

Seems like the consistent theme here is a crash as soon as any chosen is about to take lethal damage. Do we know if there's a workaround?

I think these errors below are just always there, and nothing useful, but here's the tail of my `Launch.log`:
```
[3092.22] Log: ###### OnlineEventMgr waited 0.00 seconds for saves to complete
[3106.69] Log: --- LOADING MOVIE TIME: 15.7 sec ---
[3122.48] XCom_Maps: , Plot_SHN_Retalliation_LgObj_Stream, Temperate, 21.37
[3569.41] Warning: SeqAct_IsUnitAGivenTemplate: No unit or invalid unit passed to kismet action

[3569.41] Warning: SeqAct_IsUnitAGivenTemplate: No unit or invalid unit passed to kismet action

[3569.41] Warning: SeqAct_IsUnitAGivenTemplate: No unit or invalid unit passed to kismet action

[3569.41] Warning: SeqAct_IsUnitAGivenTemplate: No unit or invalid unit passed to kismet action

[3569.41] Warning: X2Action_MarkerInterruptEnd tried to execute out of order! Trace
 Script call stack:
	State XComGame.X2Action_MarkerInterruptEnd:Executing
 


[3569.41] Warning: Attempting to resume UITimer with no active gamestate

[3569.58] Log: ###### OnlineEventMgr waited 0.00 seconds for saves to complete
[3577.71] XCom_Maps: , Plot_SHN_Retalliation_LgObj_Stream, Temperate, 4.60
[3580.26] Log: --- LOADING MOVIE TIME: 10.4 sec ---
[3821.25] Warning: X2Action_MarkerInterruptEnd tried to execute out of order! Trace
 Script call stack:
	State XComGame.X2Action_MarkerInterruptEnd:Executing
 


[3821.41] Log: ###### OnlineEventMgr waited 0.00 seconds for saves to complete
[3829.38] XCom_Maps: , Plot_SHN_Retalliation_LgObj_Stream, Temperate, 4.27
[3831.78] Log: --- LOADING MOVIE TIME: 10.1 sec ---
[4311.31] Log: ###### OnlineEventMgr waited 0.00 seconds for saves to complete
[4319.18] XCom_Maps: , Plot_SHN_Retalliation_LgObj_Stream, Temperate, 4.07
[4321.64] Log: --- LOADING MOVIE TIME: 10.1 sec ---
```
(Worth noting that I'm using more/less that Mod Jam pack of mods, so my environment isn't clean (tho at least OP's seems to have been))

---

> I've had this with killing a different Chosen, crash with SIGSEV(11) and a complaint about /lib/private_symbol_hack.so on Linux. I can confirm having experienced this bug myself now that the same save works on windows and will pass the killing of the chosen. (Which isn't a shock given the error, but nice to prove).

Foldi-One, this is present in basically all crashes we get on Linux. iirc more/less its cuz we don't have the same debug symbols present that feral does/did in their internal builds.

**Comment by AV00** [comment]
I've got the same identical issue, keeps crashing when I kill a chosen, I'm on Linux too.
Any news about it ?

**Comment by kaphula** [comment]
I don't think so, unfortunately.

**Comment by AV00** [comment]
I've done some digging of the dump: it's a null-pointer dereference, with a small (x38) offset. So someone is assuming a pointer points somewhere while it's a zero. But more than that, I can't do. I wonder if we could try to get in touch with Feral about it.

**Comment by Tedster59** [comment]
Does the crash still occur if using the beta 1.29 Highlander? I have a theory but have no real crash dump experience myself.

**Comment by AV00** [comment]
> Does the crash still occur if using the beta 1.29 Highlander? I have a theory but have no real crash dump experience myself.

I'll try but I'm not sure how to do a manual install of the mod.

**Comment by mrhyperventilate** [comment]
Just ran into this problem, and thanks to this bug report I was able to quickly track down the issue:

https://github.com/long-war-2/lwotc/pull/1806
